### PR TITLE
Dead code and cleanup.

### PR DESCRIPTION
**Pull request by Frozen**

### CODE REVIEW

File: core/blockchain_impl.go [hunk 1]
```diff
@@ -447,7 +447,7 @@ func VerifyIncomingReceipts(blockchain BlockChain, block *types.Block) error {
 			}
 		}
 
-		if err := blockchain.Validator().ValidateCXReceiptsProof(cxp); err != nil {
+		if err := NewBlockValidator(blockchain).ValidateCXReceiptsProof(cxp); err != nil {
```
Comment (MaxMustermann2) [inline core/blockchain_impl.go:450]: Would it be better to keep `bc.Validator()` without mutex as a function for cases like this?

File: core/blockchain_impl.go [hunk 2]
```diff
@@ -1764,7 +1746,7 @@ func (bc *BlockChainImpl) insertChain(chain types.Blocks, verifyHeaders bool) (i
 			err = <-verifyHeadersResults
 		}
 		if err == nil {
-			err = bc.Validator().ValidateBody(block)
+			err = NewBlockValidator(bc).ValidateBody(block)
```
Comment (MaxMustermann2) [inline core/blockchain_impl.go:1749]:
```suggestion
			err = bc.validator.ValidateBody(block)
```

File: core/staking_verifier.go [hunk 3]
```diff
@@ -30,14 +30,9 @@ var (
 )
 
 func checkDuplicateFields(
-	bc ChainContext, state vm.StateDB,
+	addrs []common.Address, state vm.StateDB,
```
Comment (MaxMustermann2) [inline core/staking_verifier.go:33]: For clarity, can you rename `addrs` to be `validatorList` ?

**Review by MaxMustermann2 — CHANGES_REQUESTED** [review]

File: core/blockchain_impl.go (hunk 1 above)
Comment (sophoah) [reply core/blockchain_impl.go:450]: @MaxMustermann2 just curious, could you explain your rationale of doing so ?

**Review by ONECasey — APPROVED** [review]

**Pull request merged**